packages/api/tests/endpoints/test-executions.js

```diff
@@ -170,3 +369,240 @@ test('GET fails if execution is not found', async (t) => {
+            },
+          },
+          {
+            bool: {
+              should: [{ match: { status: fakeGranules[0].status } }],
+              minimum_should_match: 1,
+            },
+          },
+        ],
+      },
+    },
+  };
+
+  const body = {
+    index: esIndex,
+    query: expectedQuery,
+  };
+
+  const response = await request(app)
+    .post('/executions/history')
+    .send(body)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`);
+
+  response.body.forEach((execution, index) => {
+    t.is(execution.arn, fakePGExecutions[index].arn);
+    t.is(execution.workflow_name, fakePGExecutions[index].workflow_name);
+  });
+});
+
+test.serial('POST /executions/history returns 400 when a query is provided with no index', async (t) => {
+  const expectedQuery = { query: 'fake-query' };
+
+  const body = {
+    query: expectedQuery,
+  };
+
+  const response = await request(app)
+    .post('/executions/history')
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(body)
+    .expect(400);
+
+  t.regex(response.body.message, /Index is required if query is sent/);
+});
+
+test.serial('POST /executions/history returns 400 when no IDs or query is provided', async (t) => {
+  const expectedIndex = 'my-index';
+
+  const body = {
+    index: expectedIndex,
+  };
+
+  const response = await request(app)
+    .post('/executions/history')
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(body)
+    .expect(400, /One of ids or query is required/);
+
+  t.regex(response.body.message, /One of ids or query is required/);
+});
+
+test.serial('POST /executions/history returns 400 when IDs is not an array', async (t) => {
+  const expectedIndex = 'my-index';
+
+  const body = {
+    index: expectedIndex,
+    ids: 'bad-value',
+  };
+
+  const response = await request(app)
+    .post('/executions/history')
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(body)
+    .expect(400);
+
+  t.regex(response.body.message, /ids should be an array of values/);
+});
+
+test.serial('POST /executions/history returns 400 when IDs is an empty array', async (t) => {
+  const expectedIndex = 'my-index';
+
+  const body = {
+    index: expectedIndex,
+    ids: [],
+  };
+
+  const response = await request(app)
+    .post('/executions/history')
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(body)
+    .expect(400);
+
+  t.regex(response.body.message, /no values provided for ids/);
+});
+
+test.serial('POST /executions/history returns 400 when the Metrics ELK stack is not configured', async (t) => {
+  const expectedIndex = 'my-index';
+  const expectedQuery = { query: 'fake-query' };
+
+  const body = {
+    index: expectedIndex,
+    query: expectedQuery,
+  };
+
+  process.env.METRICS_ES_USER = undefined;
```

Comment:
```suggestion
  const metricsUser = process.env.METRICS_ES_USER ;
  delete process.env.METRICS_ES_USER;
  t.teardown(() => process.env.METRICS_ES_USER = metricsUser);
```

This will prevent this test from potentially interfering with other tests that might depend on the ELK stack